code-challenges.test.js

```diff
@@ -13,40 +11,102 @@
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
-
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
-
+describe("capWords",  () =>{
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+  
+      const hitchhikersCharacters = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(capWords(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+// Good failure: ReferenceError: capWords is not defined
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//name: capWords
+//input and output: array of objects, returns array with sentences
+//Process:
+// Im going to use iteration to access the values and the name of the person
+//im going to take that name and use charAt and toUpperCase to access and modify the first letter
 
-// --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
-// a) Create a test with an expect statement using the variables provided.
+const capWords = (array) => {
+  let capitalizedSentence = array.map((person) => {
+    const name = person.name
+    const capitalizedName = name.charAt(0).toUpperCase()
```

Comment:
This variable `capitalizedName` is set to the first letter of the name string capitalized. So right now, it's only 1 letter. You can use string concatenation to add more to that string, so you can add what you want. In your research, you mentioned finding .split. If you use .split(" ") with that space in between the quotes, it will split the string at the space. So "aleja hale" becomes ["aleja", "hale"]. From here you can iterate through the array and access the first and last name separately.